src/zookeeper/zookeeper_session.cpp

```diff
@@ -158,27 +167,48 @@ int zookeeper_session::attach(void *callback_owner, const state_callback &cb)
 {
     utils::auto_write_lock l(_watcher_lock);
     if (nullptr == _handle) {
-        if (FLAGS_enable_zookeeper_kerberos) {
+        if (utils::is_empty(FLAGS_sasl_mechanisms_type)) {
+            _handle = zookeeper_init(
+                FLAGS_hosts_list, global_watcher, FLAGS_timeout_ms, nullptr, this, 0);
+        } else {
+            int err = sasl_client_init(nullptr);
```

Comment:
To improve the readability, you can reduce the indents, for example:
```
utils::auto_write_lock l(_watcher_lock);
if (nullptr != _handle) {
    return;
}

if (kerberos_is_disabled) {
   ...
   return;
}

int err = sasl_client_init(nullptr);
...
```

Reply:
Didn't update this?

Reply:
Should I repeat the following code?

https://github.com/apache/incubator-pegasus/pull/1914/files#:~:text=CHECK_NOTNULL(_handle%2C,zoo_state(_handle)%3B